airbyte-cdk/bulk/core/load/src/main/kotlin/io/airbyte/cdk/task/TaskRunner.kt

```diff
@@ -31,11 +31,19 @@ class TaskRunner {
     }
 
     suspend fun run() = coroutineScope {
-        queue.consumeAsFlow().collect { task ->
-            launch {
-                log.info { "Executing task: $task" }
-                task.execute()
+        try {
+            queue.receiveAsFlow().collect { task ->
+                launch {
+                    log.info { "Executing task: $task" }
+                    task.execute()
+                }
             }
+        } catch (t: Throwable) {
+            log.error { "Exception in task runner, flushing queue: $t" }
+            queue.receiveAsFlow().collect { task ->
```

Comment:
I'm realizing it's not clear to me how this manages to exit without a `close`. I wonder if that means the previous `receiveAsFlow` will stop after emptying the queue?

Reply:
receiveAsFlow exits when the underlying channel is closed. I don't think I understand your question

Reply:
I mean that nothing closes the channel during flush, so it should block, but it doesn't. So I'm missing something.